src/components/base/JIcon/JIcon.js

```diff
@@ -1,45 +1,51 @@
 // @flow
 
-import React from 'react'
 import classNames from 'classnames'
+import React, { PureComponent } from 'react'
+
 import { iconsUI } from 'utils/sprite'
 
 export type JIconColor = 'white' | 'blue' | 'gray' | 'sky' | 'red' | 'black'
 
 type Props = {
   name: string,
-  color: ?JIconColor,
+  color: JIconColor,
 }
 
-function JIcon({
-  name, color,
-}: Props) {
-  const iconData = iconsUI[`${name}-usage`]
-  const hasFill = name.indexOf('use-fill') !== -1
-
-  if (iconData) {
-    return (
-      <svg
-        className={classNames(
-          'j-icon',
-          color && `-${color}`,
-          hasFill && '-use-fill',
-        )}
-        width={iconData.width}
-        height={iconData.height}
-      >
-        <use xlinkHref={iconData.url} />
-      </svg>
-    )
-  } else {
-    return (
-      <div className='j-icon -no-icon' />
-    )
+class JIcon extends PureComponent<Props> {
```

Comment:
https://www.peterbe.com/plog/memo-instead-of-purecomponent

I think it's better to use React.memo instead of a PureComponent  here